File: fixtures/code_samples/go/sample.go
package main

import "fmt"

// Config holds application configuration.
type Config struct {
	Name  string
	Value int64
}

// NewConfig creates a new Config.
func NewConfig(name string, value int64) *Config {
	return &Config{Name: name, Value: value}
}

// Display returns a formatted string.
func (c *Config) Display() string {
	return fmt.Sprintf("%s: %d", c.Name, c.Value)
}

func helper(x int) int {
	return x * 2
}

func main() {
	cfg := NewConfig("test", 42)
	fmt.Println(cfg.Display())
	fmt.Println(helper(10))
}

File: fixtures/code_samples/python/sample.py
"""A sample Python module for testing the parser."""


class Config:
    """Configuration class."""

    def __init__(self, name: str, value: int):
        self.name = name
        self.value = value

    def display(self) -> str:
        return f"{self.name}: {self.value}"

    def _internal(self):
        pass


def helper(x: int) -> int:
    return x * 2


def _private_fn():
    cfg = Config("test", 42)
    result = helper(10)
    print(cfg.display())
    return result
